### PR TITLE
[rosie] sequence imrovments,  unregistered factories

### PR DESCRIPTION
**Pull request by adam187**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/rosiejs/rosie#readme>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by chrislgrigg — APPROVED** [review]

File: types/rosie/index.d.ts [hunk 1]
```diff
@@ -163,7 +165,8 @@ declare namespace rosie {
      * @param {function(number): *=} builder
      * @return {Factory}
      */
-    sequence(name: keyof T, dependenciesOrBuilder?: () => any | keyof T[], builder?: Function) : IFactory<T>;
+    sequence<K extends keyof T>(name: K, builder?: (i: number) => any): IFactory<T>;
+    sequence<K extends keyof T, D extends keyof T>(name: keyof T, dependencies: D[], builder: (i: number, ...args: any[]) => any): IFactory<T>;
```
[the remark this answers is not in the document]
Comment (adam187) [reply types/rosie/index.d.ts]: @amcasey Fixed

**Comment by adam187** [comment]
Hi @RyanCavanaugh, @andy-ms

I see that npm package updated 6 days ago but this changes are not present in this version (`0.0.35`).

https://www.npmjs.com/package/@types/rosie

**Comment by sandersn** [comment]
@adam187 Looks like the types publisher still thinks 0.0.35 is the latest version. I tried manually running our re-tagging script but it produced the same result. I'll investigate.

(I'm the new owner of types-publisher since Andy moved on to work on the .NET team.)

**Comment by sandersn** [comment]
OK, the tagging script worked after deleting the `npm info` cache. 
Looks like the cache invalidation code is wrong.

**Pull request merged**